services/housefy/src/main/java/com/dech/housefy/security/AccessFilter.java

```diff
@@ -0,0 +1,81 @@
+package com.dech.housefy.security;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.regex.Pattern;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import com.dech.housefy.dto.RoleDTO;
+import com.dech.housefy.service.IJwtService;
+import com.dech.housefy.service.IRoleService;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+
+
+@Component
+@RequiredArgsConstructor
+public class AccessFilter extends OncePerRequestFilter {
+    private final IJwtService jwtService;
+    private final UserDetailsService userDetailsService;
+    private final IRoleService roleService;
+
+    @Override
+    protected void doFilterInternal(
+            @NotNull HttpServletRequest request,
+            @NotNull HttpServletResponse response,
+            @NotNull FilterChain filterChain
+    ) throws ServletException, IOException {
+        final String methodRequest = request.getMethod();
+        final String pageRequest = request.getRequestURI();
+        final String authorizationHeader = request.getHeader("Authorization");
+    
+        if (authorizationHeader == null || !authorizationHeader.startsWith("Bearer ")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        final String token = authorizationHeader.replace("Bearer ", "");
+        final String email = jwtService.getEmailFromToken(token);
+        UserDetails user = userDetailsService.loadUserByUsername(email);
+
+        Collection<? extends GrantedAuthority> roles = user.getAuthorities();
+
+        boolean hasAccess = checkAccess(roles, pageRequest, methodRequest);
+        if(!hasAccess){
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            response.getWriter().write("Access denied");
```

Comment:
I think that you should throws an exception AccesDenied or UnauthorizedException, you can check the examples in the folder error